accounts-db/src/accounts_db.rs

```diff
@@ -6755,140 +6705,170 @@ impl AccountsDb {
+                                thread_accum.num_did_not_exist += slot_info.num_did_not_exist;
+                                thread_accum.num_existed_in_mem += slot_info.num_existed_in_mem;
+                                thread_accum.num_existed_on_disk += slot_info.num_existed_on_disk;
+                                thread_accum.lt_hash.mix_in(&slot_info.slot_lt_hash.0);
+                                num_processed.fetch_add(1, Ordering::Relaxed);
+                            }
+                            thread_accum
+                        })
+                })
+                .collect::<Result<Vec<_>, _>>()
+                .expect("spawn threads");
+            let logger_thread_handle = thread::Builder::new()
+                .name("solGenIndexLog".to_string())
+                .spawn_scoped(s, || {
+                    let mut last_update = Instant::now();
+                    loop {
+                        if exit_logger.load(Ordering::Relaxed) {
+                            break;
+                        }
+                        let num_processed = num_processed.load(Ordering::Relaxed);
+                        if num_processed == num_storages as u64 {
+                            info!("generating index: processed all slots");
+                            break;
+                        }
+                        let now = Instant::now();
+                        if now - last_update > Duration::from_secs(2) {
+                            info!("generating index: processed {num_processed}/{num_storages} slots...");
+                            last_update = now;
+                        }
+                        thread::sleep(Duration::from_millis(500))
+                    }
+                })
+                .expect("spawn thread");
+            for thread_handle in thread_handles {
+                let Ok(thread_accum) = thread_handle.join() else {
+                    exit_logger.store(true, Ordering::Relaxed);
+                    panic!("index generation failed");
+                };
+                total_accum.insert_us += thread_accum.insert_us;
+                total_accum.num_accounts += thread_accum.num_accounts;
+                total_accum.accounts_data_len += thread_accum.accounts_data_len;
+                total_accum
+                    .zero_lamport_pubkeys
+                    .extend(thread_accum.zero_lamport_pubkeys);
+                total_accum.all_accounts_are_zero_lamports_slots +=
+                    thread_accum.all_accounts_are_zero_lamports_slots;
+                total_accum
+                    .all_zeros_slots
+                    .extend(thread_accum.all_zeros_slots);
+                total_accum.num_did_not_exist += thread_accum.num_did_not_exist;
+                total_accum.num_existed_in_mem += thread_accum.num_existed_in_mem;
+                total_accum.num_existed_on_disk += thread_accum.num_existed_on_disk;
+                total_accum.lt_hash.mix_in(&thread_accum.lt_hash);
```

Comment:
feels like this should be extracted to a IndexGenerationAccumulator's function moving the rhs value, e.g. `fn accumulate(&mut self, other: Self)` or maybe even implementing `AddAssign` trait